pkcs11/signer.go

```diff
@@ -168,7 +167,12 @@ func (s *signer) SignX509Cert(ctx context.Context, cert *x509.Certificate, keyId
 		return nil, errors.New("client request timed out, skip signing X509 cert")
 	}
 	defer pool.put(signer)
-	cert.SignatureAlgorithm = getSignatureAlgorithm(signer.signAlgorithm())
+	// Validate the cert request to ensure it matches the keyType and also the HSM supports the signature algo.
+	if val := isValidCertRequest(cert, signer.signAlgorithm()); !val {
+		log.Printf("SignX509Cert: unsupported signature algo %d, supported algo %d", cert.SignatureAlgorithm, signer.signAlgorithm())
```

Comment:
can we log cert `CN` or some other field which will allow us track down misbehaving clients?